### ISSUE TITLE
wayland: error handling UWAC events

### ISSUE THREAD

**Issue by manio** [FreeRDP/FreeRDP#5439]
@akallabeth

Hi!
I have a problem with wlfreerdp running under wayland. It is reporting: "error handling UWAC events". When I connect with xfreerdp (using xwayland) it is connecting properly without any problems.

I was testing the client from debian package but then I removed those packages, compiled recent git master and installed it - unfortunately the problem still exists:
```
$ wlfreerdp /v:some_host:SOME_PORT
[15:39:06:693] [9439:9439] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[15:39:06:694] [9439:9439] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[15:39:06:694] [9439:9439] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
Username: manio
Domain:   
Password: 
[15:39:12:542] [9439:9439] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRA32
[15:39:12:542] [9439:9439] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
[15:39:12:544] [9439:9439] [INFO][com.winpr.clipboard] - initialized POSIX local file subsystem
[15:39:12:553] [9439:9445] [INFO][com.freerdp.channels.rdpsnd.client] - Loaded fake backend for rdpsn
d
[15:39:12:611] [9439:9439] [ERROR][com.freerdp.client.wayland] - error handling UWAC events

```
Here is the same connection but with /log-level:trace:
```
[15:46:10:505] [9584:9584] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[15:46:10:506] [9584:9584] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[15:46:10:506] [9584:9584] [DEBUG][com.freerdp.channels.cliprdr.client] - VirtualChannelEntryEx
[15:46:10:506] [9584:9584] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[15:46:10:513] [9584:9584] [DEBUG][com.freerdp.core.nego] - Enabling security layer negotiation: TRUE
[15:46:10:513] [9584:9584] [DEBUG][com.freerdp.core.nego] - Enabling restricted admin mode: FALSE
[15:46:10:513] [9584:9584] [DEBUG][com.freerdp.core.nego] - Enabling RDP security: TRUE
[15:46:10:513] [9584:9584] [DEBUG][com.freerdp.core.nego] - Enabling TLS security: TRUE
[15:46:10:513] [9584:9584] [DEBUG][com.freerdp.core.nego] - Enabling NLA security: TRUE
[15:46:10:513] [9584:9584] [DEBUG][com.freerdp.core.nego] - Enabling NLA extended security: FALSE
[15:46:10:513] [9584:9584] [DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_NLA
[15:46:10:513] [9584:9584] [DEBUG][com.freerdp.core.nego] - Attempting NLA security
[15:46:10:518] [9584:9584] [DEBUG][com.freerdp.core] - connecting to peer XXXXXXXX
[15:46:10:518] [9584:9584] [DEBUG][com.freerdp.core.nego] - RequestedProtocols: 3
[15:46:10:599] [9584:9584] [DEBUG][com.freerdp.core.nego] - RDP_NEG_RSP
[15:46:10:599] [9584:9584] [DEBUG][com.freerdp.core.nego] - selected_protocol: 2
[15:46:10:599] [9584:9584] [DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_FINAL
[15:46:10:599] [9584:9584] [DEBUG][com.freerdp.core.nego] - Negotiated NLA security
[15:46:10:599] [9584:9584] [DEBUG][com.freerdp.core.nego] - nego_security_connect with PROTOCOL_HYBRID
Username: manio
Domain:   
Password: 
[15:46:17:495] [9584:9584] [DEBUG][com.winpr.sspi] - InitSecurityInterfaceExA
[15:46:17:495] [9584:9584] [DEBUG][com.freerdp.core.nla] - nla_client_init 396 : packageName=Negotiate ; cbMaxToken=12256
[15:46:17:495] [9584:9584] [TRACE][com.freerdp.core.nla] -  InitializeSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
[15:46:17:495] [9584:9584] [DEBUG][com.freerdp.core.nla] - Sending Authentication Token
[15:46:18:535] [9584:9584] [DEBUG][com.freerdp.core.nla] - CredSSP protocol support 6, peer supports 2
[15:46:18:535] [9584:9584] [TRACE][com.freerdp.core.nla] - InitializeSecurityContext  SEC_I_COMPLETE_NEEDED [0x00090313]
[15:46:18:535] [9584:9584] [DEBUG][com.freerdp.core.nla] - Sending Authentication Token
[15:46:18:724] [9584:9584] [DEBUG][com.freerdp.core.gcc] - Server rdp encryption method: NONE
[15:46:18:059] [9584:9584] [DEBUG][com.freerdp.core.info] - Client Info Packet Flags = INFO_MOUSE|INFO_DISABLECTRLALTDEL|INFO_UNICODE|INFO_MAXIMIZESHELL|INFO_LOGONNOTIFY|INFO_COMPRESSION|INFO_ENABLEWINDOWSK
EY|INFO_FORCE_ENCRYPTED_CS_PDU|INFO_LOGONERRORS|INFO_MOUSE_HAS_WHEEL|INFO_NOAUDIOPLAYBACK
[15:46:18:061] [9584:9584] [DEBUG][com.winpr.timezone] - tz: Bias=-60 sn='Central European Standard Time' dln='Central European Daylight Time'
[15:46:18:196] [9584:9584] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x1f size=37 channelId=1007)
[15:46:18:196] [9584:9584] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1007)
[15:46:18:196] [9584:9584] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1007)
[15:46:18:196] [9584:9584] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x2b size=57 channelId=1007)
[15:46:18:196] [9584:9584] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x27 size=41 channelId=1007)
[15:46:18:245] [9584:9584] [DEBUG][com.freerdp.core.rdp] - recv Synchronize Data PDU (0x1F), length: 22
[15:46:18:245] [9584:9584] [DEBUG][com.freerdp.core.rdp] - recv Control Data PDU (0x14), length: 26
[15:46:18:288] [9584:9584] [DEBUG][com.freerdp.core.rdp] - recv Control Data PDU (0x14), length: 26
[15:46:18:288] [9584:9584] [DEBUG][com.freerdp.core.rdp] - recv Font Map Data PDU (0x28), length: 26
[15:46:18:288] [9584:9584] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRA32
[15:46:18:288] [9584:9584] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
[15:46:18:292] [9584:9584] [INFO][com.winpr.clipboard] - initialized POSIX local file subsystem
[15:46:18:300] [9584:9590] [INFO][com.freerdp.channels.rdpsnd.client] - Loaded fake backend for rdpsnd
[15:46:18:405] [9584:9584] [DEBUG][com.freerdp.core.update] - Primary Drawing Order [0x0a] OpaqueRect
[15:46:18:409] [9584:9584] [DEBUG][com.freerdp.core.update] - Primary Drawing Order [0x0a] OpaqueRect
[15:46:18:410] [9584:9584] [TRACE][com.freerdp.core.update] - BitmapUpdate: 52
[15:46:18:422] [9584:9584] [ERROR][com.freerdp.client.wayland] - error handling UWAC events
[15:46:18:424] [9584:9590] [DEBUG][com.freerdp.channels.rdpsnd.client] - Device already closed

```
Packages:
wayland-protocols 1.17-1
libwayland-bin 1.16.0-1

Can you help?

**Comment by akallabeth** [comment]
Which version of freerdp are you using?
Wlfreerdp is still very limited and got a lot of fixes in recent months.

**Comment by akallabeth** [comment]
You might also want to test our nightly packages for reference.

**Comment by manio** [comment]
It's a current git master version

**Comment by akallabeth** [comment]
Ok, then I'll have to check.
Wayland support is still buggy, sorry :/

**Comment by manio** [comment]
@akallabeth
Does it make sense to comment out the https://github.com/bmiklautz/debian-freerdp2/blob/master/client/Wayland/wlfreerdp.c#L354 call?

**Comment by manio** [comment]
@akallabeth Pasted the wrong repo, I just meant the handle_uwac_events()

**Comment by manio** [comment]
It is returning because of UwacDisplayDispatch() call

**Comment by manio** [comment]
update:
wl_display_flush() in UwacDisplayDispatch() is returning -1 and errno is set to "Broken pipe"

**Comment by akallabeth** [comment]
@manio ok, sorry took some time, but I´ve tested a bit with my debian (buster) machine and can not reproduce your issue.
could you add some more details about your client machine? (os, window manager, ...)

**Comment by manio** [comment]
Sure.
It's a debian buster/sid, display manager is: sway ([this](https://github.com/swaywm/sway/commit/ddad41f423db14f89f765fc134d45d1ea8a60caf) revision)

Output configuration: https://pastebin.com/QFM8sKwa
Let me know if I can help you somehow...

**Comment by kevinoid** [comment]
For what it's worth, I just encountered the same `error handling UWAC events` error with `wlfreerdp` from the `freerdp2-wayland` Debian package (version `2.0.0~git20190204.1.2693389a+dfsg1-2`) while using sway.

I was able to reproduce the error by buliding 2693389a and bisected the fix to 7610917a4 (which then hangs after `xdg_surface@22: error 3: xdg_surface has never been configured`).  Current `master` is working well for me on Debian+sway.
